sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Common/IAnalyzerConfiguration.cs

```diff
@@ -0,0 +1,29 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2018 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace SonarAnalyzer.Common
+{
+    public interface IAnalyzerConfiguration
+    {
+        string GetProjectOutput(AnalyzerOptions options);
```

Comment:
Replaceable way to override the project output directory; this could replace parts of the logic in the UtilityAnalyzer base clases.

Reply:
@valhristov can you please rename this ? @duncanp-sonar when making such statement it is always nice to come up with a suggestion for the name  : 
What about : `GetProjectOutputPath` ?